### PR TITLE
Make clear everywhere that Authorization API endpoints may be relocated

### PR DESCRIPTION
**Pull request by garethsb**
And make `revocation_endpoint` required in the JSON Schema as per 2.0. APIs.

Based on top of the "iss" branch for PR #69, since it touches some of the same file areas, so best merged after that!

### CODE REVIEW

**Comment by peterbrightwell** [comment]
Examined on call. Needs reviewing against latest head following recent merges <- @simonlo-sony

**Comment by garethsb** [comment]
@peterbrightwell, now rebased.

File: docs/2.0. APIs.md [hunk 1]
```diff
@@ -49,19 +49,19 @@ is defined in the [Discovery](3.0.%20Discovery.md) Section as the "path componen
...
-[RFC 6749][RFC-6749]
-*   **/token** - Used for requesting a token when using any of the chosen grant types, as defined by
-[RFC 6749][RFC-6749]
-*   **/revoke** - Used as the revocation endpoint for revoking currently active access tokens or refresh tokens, as
-defined by [RFC 7009][RFC-7009].
-*   **/jwks** - Used to expose a JSON Web Key Set [RFC 7517][RFC-7517] containing the public key that corresponds with
-the private key used to sign the access tokens.
-*   **/register** - Used for the manual or dynamic registration of an OAuth 2.0 client with the Authorization
-Server in line with [RFC 7591][RFC-7591].
+The following endpoints MUST be implemented by an Authorization Server at the paths signalled in the Authorization
```
Comment (prince-chrismc) [inline docs/2.0. APIs.md]: American spelling?

```suggestion
The following endpoints MUST be implemented by an Authorization Server at the paths signaled in the Authorization
```

Comment (garethsb) [reply docs/2.0. APIs.md]: 🤷‍♂️... If we have to! 
Thanks, @prince-chrismc, shows you were reviewing closely 🙂

**Comment by garethsb** [comment]
Should we indicate that the Authorization API base resource GET method is optional (`get?:`) - or allow 404 - or is it really required? (This is one of the troubles of this RAML being an NMOSified 'example' for the OAuth 2.0 spec.)

How about whether or not we should indicate in the Server Metadata API that for a specific issuer identifier you only need one or other of the two endpoints in this RAML? _Edit: now description has "when the issuer identifier has no path component" and "when the issue identifier contains a path component" respectively, so maybe that's good enough?_

**Comment by garethsb** [comment]
Hmm, raml2html doesn't seem to render the `displayName` of the `/.well-known/oauth-authorization-server/{api_selector}` endpoint in the output. Anyone know why? *Edit: Don't worry, now refactored into descriptions instead.*

**Comment by garethsb** [comment]
> Should we indicate that the Authorization API base resource GET method is optional (`get?:`) - or allow 404 - or is it really required? (This is one of the troubles of this RAML being an NMOSified 'example' for the OAuth 2.0 spec.)

Actually, would it be reasonable just to remove the base resource entirely, since this is not a typical NMOS API anyway?

**Comment by garethsb** [comment]
Base resource now gone.

**Review by andrewbonney — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: APIs/AuthorizationAPI.raml [hunk 2]
```diff
@@ -167,16 +162,23 @@ types:
...
     description: |
-      Method for requesting OAuth 2.0 Bearer Token (Authorization Code and Refresh Grants.)
+      Method for requesting OAuth 2.0 Bearer Token (Authorization Code and Refresh Grants).
 
-      **Note** - The data flow is dependant on the chosen grant type (e.g. authorization_code, refresh grant, etc.)
+      **Note**: The data flow is dependant on the chosen grant type (e.g. authorization_code, refresh grant, etc.)
 
-      **Note** - Request examples may render as JSON objects, but should be passed in as the `application/x-www-form-urlencoded`
-      MIME type, in accordance with [RFC6749](https://tools.ietf.org/html/rfc6749).
+      **Note**: Request examples may render as JSON objects, but should be passed in as the `application/x-www-form-urlencoded`
```
Comment (garethsb) [inline APIs/AuthorizationAPI.raml]:
```suggestion
      **Note**: Request examples may be rendered as JSON objects, but should be passed in as "application/x-www-form-urlencoded"
```

File: APIs/AuthorizationAPI.raml [hunk 3]
```diff
@@ -167,16 +162,23 @@ types:
...
+        [RFC 8414](https://tools.ietf.org/html/rfc8414) Section 2.
+      type: string
+      example: "token"
   post:
     description: |
-      Method for requesting OAuth 2.0 Bearer Token (Authorization Code and Refresh Grants.)
+      Method for requesting OAuth 2.0 Bearer Token (Authorization Code and Refresh Grants).
 
-      **Note** - The data flow is dependant on the chosen grant type (e.g. authorization_code, refresh grant, etc.)
+      **Note**: The data flow is dependant on the chosen grant type (e.g. authorization_code, refresh grant, etc.)
```
Comment (garethsb) [inline APIs/AuthorizationAPI.raml]:
```suggestion
      **Note**: The data flow is dependent on the chosen grant type (e.g. authorization_code, refresh grant, etc.)
```

**Pull request merged**